File: implementations/v3-bun-htmx/backend/src/services/WSService.ts
const WS_BROKER_URL = process.env.WS_BROKER_URL || "http://websocket:8080/broadcast";

export class WSService {
    static async broadcastUpdate(spotId: number, status: string = 'changed') {
        try {
            await fetch(WS_BROKER_URL, {
                method: 'POST',
                body: JSON.stringify({ event: 'update', spot_id: spotId, status })
            });
        } catch (e) {
            console.error("Failed to broadcast update", e);
        }
    }
}

File: implementations/v2-python-vue/backend/services/ws_service.py
import requests

WS_BROKER_URL = "http://websocket:8080/broadcast"

class WebSocketService:
    @staticmethod
    def broadcast_update(spot_id: int):
        try:
            requests.post(WS_BROKER_URL, json={
                "event": "update",
                "spot_id": spot_id,
                "status": "changed"
            }, timeout=1)
        except:
            pass # Fire and forget
